### ISSUE TITLE
Analytics Implementation or QA Support for [MOVE - Accessible 1095-B MVP]

### ISSUE THREAD

**Issue by tameracorson** [department-of-veterans-affairs/va.gov-team#44810]
# Google Analytics Implementation or QA Request

## What this form is for

<details>
<summary>More Information</summary>
<div>
    <br>
<p>Use this template to request Google Analytics implementation or QA for your product. Please fill out all brackets.</p>
    
> _Please see the following information for implementation timeline:_
> New Teams: Implementation will usually be completed in 4 sprints
> 
> - Sprint 1: Discovery/Orientation
> - Sprint 2: FE Implementation
> - Sprint 3: Tagging Implementation
> - Sprint 4: QA
> 
> All other teams: Implementation will usually be completed in 2 sprints
> 
> - Sprint 1: Discovery and FE Implementation
> - Sprint 2: Tagging Implementation and QA

</div>
</details>

### Required
- **Is this request for **Google Analytics implementation** or **QA**?**

     - [x] Google Analytics implementation
     - [ ] QA

- **Who is this request for?**

    - Team MOVE
    - [Product outline](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/1095b-tax-form/product/1095B-Tax-Form-Product-Outline.md) 
    
- **Provide the link(s) for Google Analytics implementation or QA**
  
    - [Insert link(s)]
    
- **When is the launch date?**
    - Staging review: 8/18/2022 
    - Dark launch 8/22/2022 (estimated)
    - Production launch: 9/1/2022  (estimated)

- **What interactions need to be tracked?** _NOTE: [Design System Components](https://depo-platform-documentation.scrollhelp.site/analytics-monitoring/Design-System-Components.1840808040.html) are automatically tracked_
    
    - [If this is a funnel, insert each step's URL - pages visible depending on the applicant's answers should be marked accordingly]

We have a React widget on a Drupal page, which is not going to work in Staging. Our eventual URL for the 1095-B page is going to be: https://www.va.gov/records/download-your-irs-1095-b
    
    - [Insert other interactions - please review the [Design System Components](https://depo-platform-documentation.scrollhelp.site/analytics-monitoring/Design-System-Components.1840808040.html) and [GA Data Dictionary](https://depo-platform-documentation.scrollhelp.site/analytics-monitoring/Google-Analytics-Data-Dictionary.1810464912.html)  to make sure these interactions aren't already tracked]

- We need to track a radio button selection. It allows users to pick between downloading a PDF or text version of the 1095-B form. 

Here's a list of all the things we want to track:
** Finding the form
- Number of users who find the page through online search (like Google)
- Number of users who search for the form on VA.gov
- Track the navigation from entering VA.gov to getting to the 1095-B page

** Accessing the form
- Page views
- % of users who land on the form and download it 
- Number (or percentage) of users who download the text form vs the PDF 

### Optional
- **If applicable, what test users are needed?**  _NOTE: Do not put staging credentials in this ticket; store or reference them in a .md file in the `va.gov-team-sensitive` repository_

    - We can not test in staging. 
- **Any additional comments**

    - [Insert comments]

> Please leave the following blank

## Acceptance criteria
- [x] All appropriate issue tagging is completed
- [x] QA has been completed
- [ ] Events have been added to `Product` and `Benefit Hub` Content Groups

## Additional information

Please refer to [Platform Collaboration Cycle](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/index.html) or the [Analytics Request Touchpoint](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/Analytics-request.1782120453.html) on Platform website for more information about the Collaboration Cycle.

**Comment by khandelwal** [comment]
@michelle-dooley This is our first time submitting as new team for 1095-B. We've got a React widget on a Drupal page. Do we have enough information here, to find sometime to learn about what needs to be done to implement the analytics.

**Comment by michelle-dooley** [comment]
@khandelwal - let me confirm if the radio button will be tracked as part of our standard web component tracking or if this will require a custom event, I will let you know as soon as I confirm.

**Comment by michelle-dooley** [comment]
@khandelwal - the radio button tracking is something you will need to opt into.  I have cc'd @jennaouellet to walk you through how to do that.

**Comment by khandelwal** [comment]
Thanks! I'll have to invite a few more folks from my end. @jennaouellet What's the best way to schedule with you?

**Comment by jennaouellet** [comment]
Hi @khandelwal! We have the radio button event documented [here](https://depo-platform-documentation.scrollhelp.site/analytics-monitoring/design-system-components). This is the code you'll need to add:

```
'event': 'int-radio-button-option-click', //consistently populate for all radio button toggles
'radio-button-label': //dynamically populate according to the radio button label for the group presented to the user
'radio-button-optionLabel': //dynamically populate according to the radio button option selected label
'radio-button-required: //dynamically populates with boolean true or false, true if radio button is required
```

Please let us know if you have any other questions. Thank you!

**Comment by stephte** [comment]
@jennaouellet how do we test locally that it is tracking properly?

**Comment by stephte** [comment]
@michelle-dooley @jennaouellet  Our goal with the radio button tracking is to send an event showing which form they downloaded, correct? In that case, we don't want to track the radio button toggle specifically, but we'd rather have an event that triggers when the 'Download' button is selected, and sending which option is downloaded to analytics.

For this case, should we send the 'cta-button-click' event instead of the 'int-radio-button-option-click' event to analytics (since now we are technically tracking the button click, not the radio toggle)?

**Comment by jonathan-epstein13** [comment]
Just posting the update here. Confirmed in the ##platform-analytics channel to use the radio-button click event because radio buttons are used to allow users to see all available choices at once and select exactly one option.

**Comment by tameracorson** [comment]
@stephte @munishmo We need this answered above on this ticket:

**1) Provide the link(s) for Google Analytics implementation or QA**
[Insert link(s)]

**2)  then answer these questions:**
    * - [ ] All appropriate issue tagging is completed
    * - [ ] QA has been completed
    * - [ ] Events have been added to Product and Benefit Hub Content Groups

Please tag me when done

**Comment by khandelwal** [comment]
@jonathan-epstein13 @michelle-dooley I've mostly read through some of the analytics guidance online. I'm a little confused. I think, it track the things we want to track, we don't need a custom dashboard for 1095-B. Is that correct? Or, based on what we have above, we should have one?

Our tracking code isn't merged in yet, but we'll eventually need to know where to look for our page in the analytics.

**Comment by jonathan-epstein13** [comment]
@khandelwal so you want to know where to find the dashboard for 1095-B? In relation to this new tracking or something else entirely? 

For the tracking, once the tracking code is merged, we can start to see it in staging or if there is another way to test, let us know.

**Comment by tameracorson** [comment]
@jonathan-epstein13 Hi Jonathan, One of the tickets asked for the board... cant remember which but...  Where do we get a link to the board?

**Comment by khandelwal** [comment]
@jonathan-epstein13 Yes! I want to know where to find the dashboard for 1095-B. We want to track all these things

- Number of users who find the page through online search (like Google)
- Number of users who search for the form on VA.gov 
- Track the navigation from entering VA.gov to getting to the 1095-B page
- Page views
- % of users who land on the form and download it
- Number (or percentage) of users who download the text form vs the PDF

I'm assuming once everything is in staging/production we'll see this somewhere? We don't have to create a dashboard? 

Sorry for the questions, my knowledge of analytics is weaker than anyone would like.

**Comment by jonathan-epstein13** [comment]
@khandelwal I am going to relay this over to the team and see what needs to get done in terms of the metrics you are looking for and the dashboard.
cc: @michelle-dooley

**Comment by khandelwal** [comment]
Thanks!

**Comment by michelle-dooley** [comment]
Hi @khandelwal - In order to get 1095-B added to the Forms KPI dashboard you will need to complete the [KPI Dashboard - Request Template](https://github.com/department-of-veterans-affairs/va.gov-team/issues/new?assignees=joanneesteban&labels=analytics-insights%2C+analytics-request%2C+kpi-dashboard&template=analytics-request-kpi-dashboard.md&title=Analytics+KPI+dashboard+request+for+%5BTeam%5D).    Here is a link to the current [forms dashboard](https://va-gov.domo.com/page/447193050) so you can see the metrics that are reported, once we get 1095-B added it would be in the form_name drop down at the top of the page for you to filter on.  I have also inlcuded some more info about our dashboards from the [platform website](https://github.com/department-of-veterans-affairs/va.gov-team/issues/new?assignees=joanneesteban&labels=analytics-insights%2C+analytics-request%2C+kpi-dashboard&template=analytics-request-kpi-dashboard.md&title=Analytics+KPI+dashboard+request+for+%5BTeam%5D)
